src/onegov/event/collections/occurrences.py

```diff
@@ -211,6 +252,51 @@ def tag_counts(self) -> defaultdict[str, int]:
 
         return counts
 
+    def valid_keywords(self, parameters):
+        return {
+            as_internal_id(k): v for k, v in parameters.items()
+            if k in {
+                as_internal_id(kw) for kw in
+                self.request.app.org.event_filter_configuration.get(
+                    'keywords', set)
+            }
+        }
+
+    def available_filters(self, sort_choices=False, sortfunc=None):
+        """
+        Retrieve the filters with their choices. Return by default in the
+        order of how they are defined in the config structure.
+        To filter alphabetically, set sort_choices=True.
+
+        :return tuple containing tuples with keyword, label and list of values
+        :rtype tuple(tuples(keyword, title, values as list)
+
+        """
+        keywords = tuple(
+            as_internal_id(k) for k in
+            self.request.app.org.event_filter_configuration.get('keywords',
+                                                                set)
+        )
+
+        fields = {
+            f.id: f for f in
+            self.request.app.org.event_filter_fields if (f.id in keywords)
```

Comment:
@Daverball Do you have a better approach to access the organization settings. This way the collection requires the request which gives me some headache in the unit tests

Reply:
I would manually pass them into the collection, rather than trying to retrieve them. This decouples the module properly from `onegov.org` anyways, since otherwise it has a soft dependency that should not be there.

This means that you will need to properly bind those parameters in the views on that collection, but I think that's fine. Alternatively you can override the `OrgApp.path` directive for this collection to automatically bind those parameters for you.